test/cache.c

```diff
@@ -32367,20 +32054,19 @@ check_write_permitted(const H5F_t H5_ATTR_UNUSED *f, bool *write_permitted_ptr)
  *****************************************************************************/
 
 H5F_t *
-setup_cache(size_t max_cache_size, size_t min_clean_size, unsigned paged)
+setup_cache(size_t max_cache_size, size_t min_clean_size, unsigned paged, H5CX_node_t *api_ctx)
 {
-    char        filename[512];
-    bool        show_progress = false;
-    bool        verbose       = true;
-    int         mile_stone    = 1;
-    hid_t       fid           = H5I_INVALID_HID;
-    H5F_t      *file_ptr      = NULL;
-    H5C_t      *cache_ptr     = NULL;
-    H5F_t      *ret_val       = NULL;
-    haddr_t     actual_base_addr;
-    hid_t       fapl_id = H5P_DEFAULT;
-    hid_t       fcpl_id = H5P_DEFAULT;
-    H5CX_node_t api_ctx = {{0}, NULL}; /* API context node to push */
```

Comment:
As `H5CX_push()` has since been refactored to use the passed in context object directly, rather then allocating a new node, this variable needs to be hoisted out of this function. Otherwise, the library will be holding onto a stale address once this function returns. The simplest way seemed to be just to pass in a pointer to a `H5CX_node_t`, but, again, this seems to point to design issues.